### ISSUE TITLE
Encoded spaces are transformed into regular spaces when logging a bug

### ISSUE THREAD

**Issue by softvision-sergiulogigan** [webcompat/webcompat.com#2836]
**Browser / Version**
Firefox Nightly 68.0a1 (2019-04-02) (Desktop)

**Operating System**
Windows 10

**Steps to reproduce:**
1. Navigate to: https://staging.webcompat.com/
2. Click on the Report bug button.
3. Fill in the "Site URL" textbox with "https://www.dropbox.com/s/5vte0zax0jkjepe/Dragon%20Ball%20Xenoverse%202%20Patch%20notes%20for%20Extra%20Pack%203%20(DLC7).pdf?dl=0".
4. Fill in/select the other options with test data.
5. Report the issue.

**Expected Behavior:**
The URL keeps the spaces encoded as `%20`.

**Actual Behavior:**
The encoded spaces are transformed into regular spaces.

**Notes:**
1. The URL's need to be copied/pasted in order to access them.
2. See https://github.com/webcompat/webcompat-tests/issues/1773.

**Watchers:**
@softvision-oana-arbuzov
@softvision-sergiulogigan
@cipriansv

sv;

**Comment by karlcow** [comment]
Thanks @softvision-sergiulogigan for reporting.

**Comment by karlcow** [comment]
so this doesn't seem to happen on the python side. 

```diff
diff --git a/tests/unit/test_form.py b/tests/unit/test_form.py
index 4ceeb73..8bae86d 100644
--- a/tests/unit/test_form.py
+++ b/tests/unit/test_form.py
@@ -67,6 +67,9 @@ class TestForm(unittest.TestCase):
         r = form.normalize_url('http://param.example.com/?q=foo#bar')
         self.assertEqual(r, 'http://param.example.com/?q=foo#bar')
 
+        r = form.normalize_url('http://example.com/there%20is%20space')
+        self.assertEqual(r, 'http://example.com/there%20is%20space')
+
         r = form.normalize_url('')
         self.assertIsNone(r)
 
@@ -194,6 +197,12 @@ class TestForm(unittest.TestCase):
         actual = form.build_formdata(form_object)
         expected = {'body': u'<!-- @browser: None -->\n<!-- @ua_header: None -->\n<!-- @reported_with: None -->\n\n**URL**: http://\u611b\n\n**Browser / Version**: None\n**Operating System**: None\n**Tested Another Browser**: Unknown\n\n**Problem type**: Unknown\n**Description**: None\n**Steps to Reproduce**:\nNone\n\n\n\n_From [webcompat.com](https://webcompat.com/) with \u2764\ufe0f_', 'title': u'\u611b - unknown'}  # noqa
         self.assertEqual(actual, expected)
+        # testing for URL containing encoded spaces %20
+        form_object = {'url': 'http://example.com/there%20is%20space'}
+        actual = form.build_formdata(form_object)
+        expected = {'body': u'<!-- @browser: None -->\n<!-- @ua_header: None -->\n<!-- @reported_with: None -->\n\n**URL**: http://example.com/there%20is%20space\n\n**Browser / Version**: None\n**Operating System**: None\n**Tested Another Browser**: Unknown\n\n**Problem type**: Unknown\n**Description**: None\n**Steps to Reproduce**:\nNone\n\n\n\n_From [webcompat.com](https://webcompat.com/) with \u2764\ufe0f_', 'title': 'example.com - unknown'}  # noqa
+        self.assertEqual(actual, expected)
+
 
     def test_get_details(self):
         """Assert we handle valid dict and other values."""
```

adding these tests didn't break anything. We need to check out the JS side of things.
hmmm I just tried…
https://github.com/webcompat/webcompat-tests/issues/1782
https://staging.webcompat.com/issues/1782

Let me try again with your specific steps.



and with the specific steps
https://staging.webcompat.com/issues/1784
https://github.com/webcompat/webcompat-tests/issues/1784


working… on macOS at least?

**Comment by softvision-sergiulogigan** [comment]
Well... Looks like I'm not able to reproduce the issue either. Tried on both Firefox and Chrome.

https://github.com/webcompat/webcompat-tests/issues/1786
https://github.com/webcompat/webcompat-tests/issues/1785

**Comment by karlcow** [comment]
Let's close :)
Thanks for re-testing.
